service.go

```diff
@@ -0,0 +1,174 @@
+package ipam
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"strings"
+	"time"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/giantswarm/microstorage"
+	"github.com/giantswarm/microstorage/memory"
+)
+
+const (
+	IPAMSubnetStorageKey       = "/ipam/subnet"
+	IPAMSubnetStorageKeyFormat = "/ipam/subnet/%s"
+)
+
+// Config represents the configuration used to create a new ipam service.
+type Config struct {
+	// Dependencies.
+	Logger  micrologger.Logger
+	Storage microstorage.Storage
+
+	// Settings.
+	// Network is the network in which all returned subnets should exist.
+	Network *net.IPNet
+}
+
+// DefaultConfig provides a default configuration to create a new ipam service
+// by best effort.
+func DefaultConfig() Config {
+	var err error
+
+	var newLogger micrologger.Logger
+	{
+		config := micrologger.DefaultConfig()
+		newLogger, err = micrologger.New(config)
+		if err != nil {
+			panic(err)
+		}
+	}
+
+	var newStorage microstorage.Storage
+	{
+		config := memory.DefaultConfig()
+		newStorage, err = memory.New(config)
+		if err != nil {
+			panic(err)
+		}
+	}
```

Comment:
I opt for not setting dependencies in `DefaultConfig`. This should be used to set default settings imo.

I think this is a good example: https://github.com/giantswarm/tprstorage/blob/master/storage.go#L44-L64

I don't think we should fall back to the default K8s client. This may generate more problems than goodness. Same goes for any dependency.